site/app/controllers/pdf/PDFController.php

```diff
@@ -157,24 +151,14 @@ public function showGraderPDFEmbedded($gradeable_id) {
             }
         }
         $active_version = $graded_gradeable->getAutoGradedGradeable()->getActiveVersion();
-        $annotation_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable_id, $id, $active_version);
+        $annotation_dir = preg_replace('/\\.[^.\\s]{3,4}$/', '', FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable_id, $id, $active_version));
```

Comment:
what is this regex doing?

Reply:
The problem with the restructure is trying to render the annotations with that format leads to the following error. It looks to mean like its somewhere in the annotate.js library
![image](https://user-images.githubusercontent.com/11095297/84004183-2bb91000-a939-11ea-8786-7700f7921048.png)

Reply:
I'm unsure what the regex does. It was in master, but seems unneeded. I've removed it.

Reply:
It's to remove the extension. The regex is roughly literal period + 3-4 non-whitespace characters.

Anyway, this regex is in a couple of other places and I think you can remove it in each place since the path is never going to have an extension we want to remove.